### PR TITLE
Use DarwinToolchain for apple-none targets

### PR DESCRIPTION
**Pull request by rauhul**
- When compiling for os none for apple, we should perfer using a Darwin toolchain (maybe a misnomer now) instead of a generic UNIX toolchain. This results in the driver selecting relevant tools for apple platforms, such as using ld64 over gold, etc.

### CODE REVIEW

**Review by artemcm — APPROVED** [review]
Could you please add a quick test for this? 
Have two driver invocations, one for `apple` one not, and make sure the corresponding drivers have the expected toolchain.

File: Sources/SwiftDriver/Driver/Driver.swift [hunk 1]
```diff
@@ -2954,12 +2954,19 @@ extension Triple {
...
       return GenericUnixToolchain.self
-    case .freeBSD, .haiku, .openbsd, .noneOS:
+    case .freeBSD, .haiku, .openbsd:
       return GenericUnixToolchain.self
     case .wasi:
       return WebAssemblyToolchain.self
     case .win32:
       return WindowsToolchain.self
+    case .noneOS:
+        switch self.vendor{
```
Comment (compnerd) [inline Sources/SwiftDriver/Driver/Driver.swift:2964]:
```suggestion
        switch self.vendor {
```

**Review by compnerd — APPROVED** [review]
LGTM with an associated test (as @artemcm already requested).

File: Sources/SwiftDriver/Driver/Driver.swift (hunk 1 above)
Comment (rauhul) [reply Sources/SwiftDriver/Driver/Driver.swift]: Picked up this diff with the latest commit that adds tests. I used a large list of triples already used in TripleTests, but I'm not sure if this is really the best test structure and I'm happy to change it if you have any ideas.

Comment (artemcm) [reply Sources/SwiftDriver/Driver/Driver.swift]: I appreciate the thoroughness. It looks good to me, thanks!

**Comment by artemcm** [comment]
@swift-ci test

**Comment by rauhul** [comment]
@swift-ci test windows

**Comment by rauhul** [comment]
@swift-ci test windows

**Comment by rauhul** [comment]
@compnerd The windows test is failing with when trying to clone `https://github.com/apple/swift-crypto.git` any ideas on how to get past this point?

**Review by kubamracek — APPROVED** [review]

**Pull request merged**